### PR TITLE
[Swift3] escape URL parameters

### PR DESCRIPTION
**Pull request by xiaochow**
### PR checklist

- [X ] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X ] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This PR is to fix the following issue #6400 for Swift4 generator. The change was originally made by @ehyche in #6705, this PR just applied it to Swift4.

The problem was that URL parameters with spaces or some other characters could lead to invalid URL because they were not escaped. In the API template of this PR, URL parameters are escaped with addingPercentEncdoing method before the generation of URLString.

This PR includes the following changes:

1. API template change and auto-generated code change in Petstore example for Swift4.
2. A unit test in UserAPITest.swift to verify the change fixes the problem.
3. Four constants in RequestBuilder were made to public for the unit testing.
4. Some other changes due to the running of bin/swift4-petstore.sh.

@jaz-ah @jgavris @Edubits @ehyche , please review. @ehyche, thank you for your original work and PR for Swift3 generator.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

File: modules/swagger-codegen/src/main/resources/swift4/APIs.mustache [hunk 1]
```diff
@@ -16,10 +16,10 @@ open class {{projectName}}API {
 open class RequestBuilder<T> {
     var credential: URLCredential?
     var headers: [String:String]
-    let parameters: [String:Any]?
-    let isBody: Bool
-    let method: String
-    let URLString: String
+    public let parameters: [String:Any]?
```
Comment (jaz-ah) [inline modules/swagger-codegen/src/main/resources/swift4/APIs.mustache]: was this a required part of this change?

File: samples/client/petstore-security-test/swift/SwaggerClient/Classes/Swaggers/AlamofireImplementations.swift [hunk 2]
```diff
@@ -66,12 +66,16 @@ class AlamofireRequestBuilder<T>: RequestBuilder<T> {
                         switch v {
                         case let fileURL as NSURL:
                             mpForm.appendBodyPart(fileURL: fileURL, name: k)
+                            break
```
Comment (jaz-ah) [inline samples/client/petstore-security-test/swift/SwaggerClient/Classes/Swaggers/AlamofireImplementations.swift]: these breaks aren't necessary - let's take them out to make the code more readable.

File: modules/swagger-codegen/src/main/resources/swift4/APIs.mustache (hunk 1 above)
Comment (xiaochow) [reply modules/swagger-codegen/src/main/resources/swift4/APIs.mustache]: It's not required in the real use, this is mainly for the testing case added to the tests.

File: samples/client/petstore-security-test/swift/SwaggerClient/Classes/Swaggers/AlamofireImplementations.swift (hunk 2 above)
Comment (xiaochow) [reply samples/client/petstore-security-test/swift/SwaggerClient/Classes/Swaggers/AlamofireImplementations.swift]: Yes I agree. These were in the old script and got generated when I ran it. I just took them out. Thanks for the checking!

**Comment by xiaochow** [comment]
@wing328 Thanks for the reminder! I'm fairly new to this so I didn't know my account was not linked to my computer. It's okay for this time, I will make sure it gets linked next time.

File: modules/swagger-codegen/src/main/resources/swift4/APIs.mustache (hunk 1 above)
Comment (jaz-ah) [reply modules/swagger-codegen/src/main/resources/swift4/APIs.mustache]: @ehyche any concerns about making these public?  we could just limit it to URLString which is used in this new test.

**Comment by wing328** [comment]
@xiaochow 👌 

If no one has further question or feedback on this PR, I'll merge it on coming Wednesday.

**Comment by wing328** [comment]
Tested locally and the result is good

**Pull request merged**